notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/filter/ReadStateSupportFilter.java

```diff
@@ -78,46 +82,48 @@ public NotificationResponse doFilter(HttpServletRequest request, INotificationSe
 
         final NotificationResponse rslt = response.cloneIfNotCloned();
 
-        // Add and implement the read behavior with our copy
-        for (NotificationCategory category : rslt.getCategories()) {
+        if (!disableDefaultAction) {
+            // Add and implement the read behavior with our copy
+            for (NotificationCategory category : rslt.getCategories()) {
 
-            for (NotificationEntry entry : category.getEntries()) {
+                for (NotificationEntry entry : category.getEntries()) {
 
-                /*
-                 * Participation in READ behavior is 100%
-                 * dependant on having an id set on the entry.
-                 */
-                if (StringUtils.isBlank(entry.getId())) {
-                    continue;
-                }
+                    /*
+                     * Participation in READ behavior is 100%
+                     * dependant on having an id set on the entry.
+                     */
+                    if (StringUtils.isBlank(entry.getId())) {
+                        continue;
+                    }
```

Comment:
Thanks for adding a doc comment here, it really helps clarify how this works :+1: